File: string/sorting-the-sentence/sorting-the-sentence.go
package sortingthesentence

import "strings"

func sortSentence(s string) string {
	words := strings.Split(s, " ")

	sorted := make([]string, len(words), len(words))

	for _, word := range words {
		lastIndex := len(word) - 1
		i := word[lastIndex]
		sorted[i-'1'] = word[:lastIndex]

	}

	return strings.Join(sorted, " ")
}

File: string/sorting-the-sentence/sorting-the-sentence.ts
function sortSentence(s: string): string {
    const words = s.split(" ");

    const sentence = Array<string>(words.length);

    for (let word of words) {
        const i = word[word.length - 1];
        sentence[parseInt(i, 10) - 1] = word.slice(0, -1);
    }

    return sentence.join(" ")
};
